pkg/haproxy/haproxy.go

```diff
@@ -333,48 +370,48 @@ func (h *HAProxyManager) run() {
 }
 
 // Reload rewrites the configuration and sends a signal to HAProxy to initiate the reload
-func (h *HAProxyManager) Reload(ports []uint16) error {
+func (h *HAProxyManager) Reload(podIPs []string, targetPort, servicePort string) error {
 	// compare ports and do nothing if they are the same
-	if reflect.DeepEqual(ports, h.ports) {
+	if targetPort == h.targetPort && servicePort == h.servicePort && reflect.DeepEqual(podIPs, h.podIPs) {
```

Comment:
are the podIPs guaranteed to be ordered?

Reply:
now sorted on instantiation of the haproxy config in realserver:::configure